repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -145,7 +147,11 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
             case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
             case Failure(e: OverLimitException) =>
               response.addHeader(HttpHeaders.RETRY_AFTER, e.retryAfter)
-              Reject(HttpServletResponse.SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
+              if (isSelfValidating) {
+                Reject(e.statusCode, Some(e.getMessage))
+              } else {
+                Reject(HttpServletResponse.SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
```

Comment:
All of the `HttpServletResponse._` are already imported. So there is no need to explicitly reference them. This isn't exactly part of this change, but they should probably be fixed.